lib/charms/opensearch/v0/helper_cluster.py

```diff
@@ -105,3 +108,38 @@ def nodes_count_by_role(nodes: List[Node]) -> Dict[str, int]:
                 result[role] += 1
 
         return result
+
+
+class ClusterState:
+    """Class for getting cluster state info."""
+
+    @staticmethod
+    @retry(
+        stop=stop_after_attempt(3),
+        wait=wait_exponential(multiplier=1, min=2, max=10),
+        reraise=True,
+    )
+    def shards(opensearch: OpenSearchDistribution, host: str = None) -> List[Dict[str, str]]:
+        """Get all shards of all indexes in the cluster."""
+        return opensearch.request("GET", "/_cat/shards", host=host)
+
+    @staticmethod
```

Comment:
nit.

What was your reasoning behind having these as static methods? they both take class, `busy_shards_by_unit` takes `self`, could we initialise `ClusterState` with `host` arg, and have these as properties?
No real downside, just curious.

Reply:
The reason I I put those as statics, is because they're self-contained helper functions that only depend on the input. The host here can be dynamic (when `None` it's referring to the current unit's IP, and can be set to use the host of an other unit such as the leader